doc/rst/platforms/comm-layers/ugni.rst

```diff
@@ -0,0 +1,228 @@
+.. _readme-ugni:
+
+======================
+Using Chapel with ugni
+======================
+
+Chapel supports a Cray-specific ``ugni`` communication layer. The ugni
+communication layer interacts with the system's network interface very closely
+through a lightweight interface called uGNI (user Generic Network Interface).
+On Cray XC systems the ugni communication layer is the default.
+
+
+Using the ugni Communications Layer
+___________________________________
+
+To use ugni communications:
+
+#. Leave your ``CHPL_COMM`` environment variable unset or set it to
+   ``ugni``:
+
+   .. code-block:: sh
+
+      export CHPL_COMM=ugni
+
+   This specifies that you wish to use the Cray-specific communication
+   layer.
+
+
+#. *(Optional)* Load an appropriate ``craype-hugepages`` module.  For example::
+
+     module load craype-hugepages16M
+
+   The ugni communication layer can be used with or without so-called
+   *hugepages*.  Performance for remote variable references is much
+   better when hugepages are used.  The only downside of using hugepages
+   is that the tasking layer may not be able to detect task stack
+   overflows by means of guard pages (see below).
```

Comment:
Would it make sense to make this a link?

Reply:
What would it link to?